folium/features.py

```diff
@@ -193,22 +195,41 @@ def __init__(self, data):
         >>> # Providing string.
         >>> GeoJson(open('foo.json').read())
 
+        >>> # Providing a style_function that put all states in green, but Alabama in blue.
+        >>> style_function=lambda x: {'fillColor': '#0000ff' if x['properties']['name']=='Alabama' else '#00ff00'}
+        >>> GeoJson(geojson, style_function=style_function)
         """
         super(GeoJson, self).__init__()
         self._name = 'GeoJson'
         if 'read' in dir(data):
-            self.data = data.read()
+            self.data = json.load(data)
         elif type(data) is dict:
-            self.data = json.dumps(data)
-        else:
             self.data = data
+        else:
+            self.data = json.loads(data)
+
+        if 'features' not in self.data.keys():
+            # Catch case when GeoJSON is just a single Feature or a geometry.
+            if not (isinstance(self.data, dict) and 'geometry' in self.data.keys()):
+                # Catch case when GeoJSON is just a geometry.
+                self.data = {'type' : 'Feature', 'geometry' : self.data}
+            self.data = {'type' : 'FeatureCollection', 'features' : [self.data]}
+
+        self.style_function = style_function if style_function is not None else (lambda x: {})
```

Comment:
Three remarks (before @ocefpaf will pep8 this) :smile: 

- the ( ) are redundant around the lambda function, they should go away;
- besides the fact this line is well over 80 lines it's anyways better to make it explicit:
```Python
if not style_function:
    style_function = lambda x: {}
self.style_funciton = style_function
```
- Finally, why did you not use a default a lambda function in the method signature? Is there some Python gotcha I'm not aware of, such as the default [] / {}? Unless there is, it's the cleanest way to do this.